### PR TITLE
Set randomAccess=true on LOAD.

### PR DESCRIPTION
**Pull request by jpountz**

### CODE REVIEW

File: lucene/core/src/java/org/apache/lucene/store/IOContext.java [hunk 1]
```diff
@@ -90,6 +90,9 @@ private IOContext(boolean readOnce, boolean load, boolean randomAccess) {
     if (readOnce && randomAccess) {
       throw new IllegalArgumentException("cannot be both readOnce and randomAccess");
     }
+    if (load && randomAccess == false) {
+      throw new IllegalArgumentException("cannot be load but not randomAccess");
```
Comment (ChrisHegarty) [inline lucene/core/src/java/org/apache/lucene/store/IOContext.java:94]: This looks right to me. ++

Comment (ChrisHegarty) [reply lucene/core/src/java/org/apache/lucene/store/IOContext.java:94]: Trivially, maybe add something like this to the javadoc for `load`:

  `If this flag is set, then randomAccess will be true.`

Comment (jpountz) [reply lucene/core/src/java/org/apache/lucene/store/IOContext.java:94]: Good call, done.

**Comment by jpountz** [comment]
@uschindler Would you like to take a look?

**Comment by uschindler** [comment]
I fixed the conflict. Sorry for force-pushing the main PR. I was not aware that you were working.

**Review by uschindler — APPROVED** [review]

**Pull request merged**